cmd/atlascmd/migrate/migrate.go

```diff
@@ -11,27 +11,79 @@ import (
 	"ariga.io/atlas/cmd/atlascmd/migrate/ent/revision"
 	"ariga.io/atlas/sql/migrate"
 	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlclient"
 	"entgo.io/ent/dialect/sql"
 	entschema "entgo.io/ent/dialect/sql/schema"
 )
 
-// A EntRevisions provides implementation for the migrate.RevisionReadWriter interface.
-type EntRevisions struct{ c *ent.Client }
+type (
+	// A EntRevisions provides implementation for the migrate.RevisionReadWriter interface.
+	EntRevisions struct {
+		ac     *sqlclient.Client // underlying Atlas client
+		ec     *ent.Client       // underlying Ent client
+		schema string            // name of the schema the revision table resides in
+	}
+
+	// Option allows to configure EntRevisions by using functional arguments.
+	Option func(*EntRevisions) error
+
+	// SchemaChanger is implemented by a driver if it how to change the connection URL to represent another schema.
+	// It will also enable EntRevisions to store the revision table in an extra schema.
+	SchemaChanger interface {
+		ChangeSchema(*sqlclient.URL, string) *sqlclient.URL
+	}
+)
 
-// NewEntRevisions creates a new EntRevisions with the given ent.Client.
-func NewEntRevisions(db schema.ExecQuerier, dialect string, opts ...ent.Option) *EntRevisions {
-	opts = append(opts, ent.Driver(sql.NewDriver(dialect, sql.Conn{ExecQuerier: db})))
-	return &EntRevisions{ent.NewClient(opts...)}
+// NewEntRevisions creates a new EntRevisions with the given ec.Client.
+func NewEntRevisions(ac *sqlclient.Client, opts ...Option) (*EntRevisions, error) {
```

Comment:
Before creating the `ent.Client`, ensures the schema exists in the database, or create it otherwise.

```go
if r.schema == "" {
	r.schema = "atlas_schema_revisions"
}
if err := client.ApplyChanges(ctx, []schema.Change{&schema.AddSchema{...}}); err != nil {
	return nil, err
}
c, err := sqlclient.Open(ctx, sqlclient.OpenSchema(r.schema))
if err != nil {
	return nil, err
}
drv := entsql.OpenDB(c.URL.Scheme, c.DB)
r.ec = ent.NewClient(ent.Driver(drv))
return r, nil
```

I'm not quite sure about `c.URL.Scheme` in SQLite. I'll check this.